test/getGitWorkspaceState.test.ts

```diff
@@ -0,0 +1,73 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import * as fse from 'fs-extra';
+import { join } from 'path';
+import { Uri } from "vscode";
+import { parseError } from 'vscode-azureextensionui';
+import { getGitWorkspaceState, GitWorkspaceState, promptForDefaultBranch, tryGetDefaultBranch, verifyGitWorkspaceForCreation } from "../extension.bundle";
+import { createTestActionContext, testFolderPath, testUserInput } from "./global.test";
+
+suite('Workspace Configurations for SWA Creation', function (this: Mocha.Suite): void {
```

Comment:
Seems like there's a few cases inside `verifyGitWorkspaceForCreation` that are not covered here
- I don't have admin access to a repo
- I have a dirty repo
- My repo is fine and no prompts/errors should occur